docs/WitnessMatchingManifesto.md

```diff
@@ -0,0 +1,210 @@
+# Protocol Witness Matching Manifesto
+
+Author: [Suyash Srijan](https://www.github.com/theblixguy) 
+
+## Introduction
+
+In Swift, protocol witnesses have to match exactly with a protocol requirement. For example:
```

Comment:
Maybe throw in "at the time of this writing", since we're talking about changing it.

The term "witness" is useful, but you need to explain it.  Just a short paragraph that sets things up for the reader, maybe with an obvious example where the witness and requirement look exactly the same.

Also, the separation of the initial absolute statement from the exceptions is just begging for a misunderstanding.  The reader will figure it out eventually, but it's always good for writing to be "tail-recursive": you should say one thing and have that lead naturally into the next thing (perhaps an example or a follow-up statement) without the reader having to remember that you were in the middle of saying something.  In this case, I'd recommend just talking about the current rules and then bringing up examples of them.  You can bring up little examples as you go along, or you can have a big block that goes through the examples later; either can work, although my personal preference is for the former if the examples are brief enough.

Don't feel like your examples need to be "real world".  Tiny examples with no distractions can be very effective at elucidating your point, e.g.:

> - Throwing requirements can be implemented by non-throwing witnesses:
>
> ```
>  protocol  P {
>     func foo() throws
>  }
>  struct S: P {
>    func foo() // allowed
>  }
>  ```